src/client/proxy/http.cc

```diff
@@ -220,47 +274,71 @@ void HTTPProxyHandler::HandleStreamRequestComplete(
           stream);
     GetOwner()->AddHandler(connection);
     connection->I2PConnect(
-        reinterpret_cast<const std::uint8_t*>(m_Request.data()),
-        m_Request.size());
+        reinterpret_cast<const uint8_t*>(m_Protocol.m_Request.c_str()),
+        m_Protocol.m_Request.size());
     Done(shared_from_this());
   } else {
     LOG(error) << "HTTPProxyHandler: stream is unavailable, try again soon";
-    HTTPRequestFailed(status_t::service_unavailable);
+    m_Protocol.m_ErrorResponse
+        = HTTPResponse(HTTPResponseCodes::status_t::service_unavailable);
+    HTTPRequestFailed();
   }
 }
-
-void HTTPProxyHandler::SetState(
-    const HTTPProxyHandler::State& state) {
-  m_State = state;
-}
-
-bool HTTPProxyHandler::CreateHTTPRequest(
-    std::uint8_t* buf,
-    std::size_t len) {
-  if (!ExtractIncomingRequest())
+/// @brief all this to change the useragent
+/// @param len length of string
+bool HTTPProtocol::CreateHTTPRequest() {
+  if (!ExtractIncomingRequest()) {
+    // m_ErrorResponse is set in ExtractIncomingRequest
     return false;
+  }
   HandleJumpService();
   // Set method, path, and version
   m_Request = m_Method;
   m_Request.push_back(' ');
   m_Request += m_Path;
   m_Request.push_back(' ');
   m_Request += m_Version + "\r\n";
-  // Set Host:
-  m_Request += m_Host + "\r\n";
-  // Reset/scrub User-Agent:
-  m_UserAgent = "MYOB/6.66 (AN/ON)";
-  m_Request += "User-Agent: " + m_UserAgent + "\r\n";
-  // Append remaining original request
-  m_Request.append(reinterpret_cast<const char *>(buf), len);
+
+  // find and remove/adjust headers
+  auto it_user_agent = std::find_if(
+      m_HeaderMap.begin(),
+      m_HeaderMap.end(),
+      [](std::pair<std::string, std::string> arg) {
+        boost::trim_left(arg.first);
+        boost::trim_right(arg.first);
+        return arg.first == "User-Agent";
+      });
+  if (it_user_agent != m_HeaderMap.end())  //  found
+    it_user_agent->second = " MYOB/6.66 (AN/ON)";
```

Comment:
>" MYOB/6.66 (AN/ON)"

Watch out for the whitespace in strings, regardless of if they are trimmed or not.